### DISCUSSION TITLE
Сюжетные вехи

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#366] [Ideas]
мне кажется надо еще одну веху истории добавить в 3 главау, что то типа предстарта 3 главы которая соответствовала бы

```
	CreateInvItems(self,ItWr_PaladinLetter_MIS,1);
	B_GiveInvItems(self,other,ItWr_PaladinLetter_MIS,1);
	KnowsPaladins_Ore = TRUE;
	B_LogEntry(Topic_MISOLDWORLD,"Командующий Гаронд дал мне письмо. Его должно быть достаточно для подтверждения моих слов. Я могу отнести его лорду Хагену.");
	MIS_ScoutMine = LOG_SUCCESS;
	MIS_ReadyForChapter3 = TRUE;
	B_NPC_IsAliveCheck(OLDWORLD_ZEN);
```

это позволит лучше видеть ситуации(ошибки, баги, сюжетные моменты) при загрузке мира и старте 3 главы

**Comment by dosinabox** [comment]
Наверное, следует блокировать возможность устанавливать у помощника уже прошедшие события (двигаться назад по сюжету), потому что в некоторых случаях это гарантированно ведет к невозможности дальнейшего прохождения.
Также следует расширить перечень "вех" у каждой главы.

**Comment by Kor-Angar** [comment]
про вехи это факт :-)  про прошлое хз, с точки зрения логики да, но это всё таки марвин и может человеку очень надо :-) а тут такие ограничения. можно предупреждение делать перед переключением о том что сюжет игры сломается.

**Comment by dosinabox** [comment]
А давай распишем все вехи сюжета.

Глава 1:
- получен доступ в город
- получен доступ в верхний квартал
- получен доступ к Хагену

Глава 2:
- принятие в Кольцо Воды
- орнамент собран
- начало квеста Гаронда
- собрана информация для Гаронда
- получено письмо для Хагена

Глава 3:
- Ворон убит
- Ватрас может покинуть город
- Глаз восстановлен

Глава 4:
- ...
- ...
- ...

Вехи первой главы вынесла в отдельные функции, которые можно будет вызывать у помощника (B_CityMainGatesPass(), B_UpperCityPass(), B_CityHallPass()).

**Reply by Kor-Angar** [reply]
Глава вторая:
Вступление В КВ, кольцо, броня и квест Ватраса
Три Информации о Рудокопах
Смерть Равена

**Comment by dosinabox** [comment]
Возможно, всё связанное с КВ и Яркендаром стоит выносит в отдельный список между 1 и 2 (или 2 и 3) главами

**Comment by dosinabox** [comment]
или в отдельный список в рамках 2 главы

**Comment by Kor-Angar** [comment]
> всё связанное с КВ и Яркендаром

сейчас это можно добавить во вторую главу, а потом продублировать, в третей если ты решишь прорабатывать третью. 

Просто отдельно выносить не очень хорошо, так как сюжет может багануться, если КВ активировать в 1 главе. если же ставить между 2 и 3 и при этом автоматом включать вторую главу, то.. фиг знает короче.

я предлагаю сделать так: дублировать кв во второй и третей главе, а когда они полностью будут сделаны, подумать как вынести их отдельно, если в этом будет необходимость. 

***

у тебя у CH есть "гильдия" КВ ?

**Comment by Kor-Angar** [comment]
> Глава вторая:
> Три Информации о Рудокопах

DIA_Garond_Success_Info

**Comment by dosinabox** [comment]
> А давай распишем все вехи сюжета.

Вот в Г1 помощник образцово-показательный:

![Screenshot_1](https://user-images.githubusercontent.com/39790055/106389287-9e5c8b80-63f3-11eb-82a7-d2e3198000d8.jpg)

**Comment by Kor-Angar** [comment]
Он прекрасен. Кстати было бы неплохо поменять внешний вид помощника по сюжету и по персонажу. Я у себя сделал помощника по персонажу Диего, а помощника по сюжету Ксардаса. Можно использовать так же и героя но в разных доспехах.

**Comment by Kor-Angar** [comment]
а чётвртую главу в Хориносе ты не убирала? :-) или я как то косо поставил обновление? :-)

![20210201213946_1](https://user-images.githubusercontent.com/49514764/106503251-72fe9d00-64d6-11eb-9183-69d5ff0c31b6.jpg)

**Comment by Kor-Angar** [comment]
я могу протестить что нибудь в любой конфигурации настроек, если есть такая потребность :-)

**Comment by dosinabox** [comment]
> а чётвртую главу в Хориносе ты не убирала? :-) или я как то косо поставил обновление? :-)

По умолчанию включено, что ли? Должно быть выключено. Но сам переключатель пока остается. Потом уберу, наверное.

> я могу протестить что нибудь в любой конфигурации настроек, если есть такая потребность :-)

Пожалуй, необходимость проверки есть только у полного удаления NPC из мира и показа заголовков при обновлении дневника.

**Comment by Kor-Angar** [comment]
> полного удаления NPC из мира

так я с этой опцией постоянно играю :-) там всё ровно.

**Comment by Kor-Angar** [comment]
> я могу протестить что нибудь в любой конфигурации настроек, если есть такая потребность :-)

ну тогда я потестирую ваш альтернативный баланс :-) где можно скачать файл g2a_nr_scriptpatch_v25_wasteland ?

**Comment by dosinabox** [comment]
[g2a_nr_scriptpatch_v25_wasteland.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/5912777/g2a_nr_scriptpatch_v25_wasteland.zip)

**Comment by dosinabox** [comment]
> ну тогда я потестирую ваш альтернативный баланс :-)

Это какой?

**Comment by Kor-Angar** [comment]
С трусость все плохо. Малые гоблины бегут от кваларона и начинается ад, так как бандит на них агрится 😀

**Reply by Kor-Angar** [reply]
> Ну, поэтому по умолчанию и выключено)

Так оно получается в целом не работает 😀 в начале игры, одно ломает, в серидине другое.

**Reply by Kor-Angar** [reply]
> С трусость все плохо. Малые гоблины бегут от кваларона и начинается ад, так как бандит на них агрится

может лучше снять трусость с молодых животных, так как они появляются в начале игры и своим убеганием не туда просто будут ломать игру. опять же разве хищники (волки, снепперы) должны убегать?

**Comment by Kor-Angar** [comment]
> уровень героя / навык 1р / итоговый ущерб

я же в итоге таки собрался все данные в одну более менее понятную таблицу. будет что с чем сравнить
https://steamcommunity.com/sharedfiles/filedetails/?id=2379834530
тут в спойлере https://rpgrussia.com/resources/rasxody-lp-na-prokachku-lajtovyj-ment-pal-c-1r.3541/

**Comment by Kor-Angar** [comment]
```
func int StoryHelper_INFO6_Condition()
{
	return TRUE;
};
```

> Главу 6 оттуда можно убрать вообще. Такая смена вне Ирдората ничего не меняет, а в самом Ирдорате не требуется.

**Reply by dosinabox** [reply]
Главу 6 оттуда можно убрать вообще. Такая смена вне Ирдората ничего не меняет, а в самом Ирдорате не требуется.

**Reply by Kor-Angar** [reply]
Почему же? 
ты же на 6 главе можешь сделать телепорт героя в каюту капитана и произойдёт смена мира.

**Reply by dosinabox** [reply]
Смена мира из каюты возможна и в первой главе.
Шестая глава начинается уже после перехода в Ирдорат, при движении на трапе (там триггер стоит).

**Reply by Kor-Angar** [reply]
значит при нажатии "6 глава" надо делать
unc void StoryHelper_KAPITEL6ANFANG()
{
   StoryHelper_KAPITEL5ANFANG();
   телепорт в каюту
};

**Reply by dosinabox** [reply]
Точно)

**Reply by dosinabox** [reply]
Но возможен ли телепорт в режиме диалога...

**Reply by Kor-Angar** [reply]
> StoryHelper_KAPITEL5ANFANG 

там  же есть выход из диалога :-)

**Reply by Kor-Angar** [reply]
ну или их можно обоих телепортировать :-) 
или еще что то. надо пробовать

**Reply by dosinabox** [reply]
ГГ можно телепортировать из диалога, но из режима диалога он выходит некорректно. Не критично, но и триггер перехода не находится прямо на точке SHIP_IN_14 и мгновенной смены мира не происходит, приходится сделать шаг вперед.

Высказывалась идея, что путь до Ирдората должен занимать какое-то время. Сейчас ГГ переходит в Ирдорат с тем же временем, что было в Хоринисе (мгновенно). Можно сделать как в Яркендаре и немного переметывать время вперед. Какое время должен занимать путь? В какое время суток приплывает корабль?

**Reply by Kor-Angar** [reply]
Про корабль мы точно говорили уже. Можно попробовать поискать :-) По моему я предлагал делать его приход на следующий день в обед.  Так как корабль идёт  не час и не два :-)

**Reply by Kor-Angar** [reply]
Главное что бы герой оказался внутри каюты, а оттуда без ключа  выход один на ирдорат:-) И в таком случае нужно ещё определиться со ставом команды.  У нас про команду отдельный топик есть. Телепортировать можно и героя и помощника по сюжету :-)

**Reply by dosinabox** [reply]
> ну или их можно обоих телепортировать :-)

Так не работает.

**Reply by Kor-Angar** [reply]
можно конечно сделать телепорт через триггер функцию. но не уверен что стоит так замарачиваться.

**Reply by dosinabox** [reply]
> Главное что бы герой оказался внутри каюты, а оттуда без ключа выход один на ирдорат:-) И в таком случае нужно ещё определиться со ставом команды.

Это больше подходит в последнюю опцию 5 главы: "Готов к отплытию". Команда набрана, капитан найден, ключ от каюты есть, доступ на корабль получен. Отдельная опция 6 главы не имеет смысла и работает криво.

**Reply by Kor-Angar** [reply]
по смыслу может быть не понятно и не очевидно. тогда в 6 можно просто продублировать готов к отплытию.

опять же в 6 так же надо сделать и финальную битву с драконом.

**Reply by dosinabox** [reply]
> По моему я предлагал делать его приход на следующий день в обед.

Тогда нападение орков произойдет примерно ночью. Мне нравится)

**Reply by Kor-Angar** [reply]
в рамках "готов к отплытию" можно сделать выбор экипажем героем. то есть дать возможность самомстоятельно выбрать и капитана и команду.

**Reply by Kor-Angar** [reply]
можно пойти еще дальшь и сделать опционально:
"быстрое отплытие" и "набор команды"

в "быстрое отплытие", закинуть npc из Готики 1 :-)
Капитан Торлоф + команда:
•	Мильтен
•	Диего
•	Лестер
•	Ли 
•	Ларес
•	Волк
•	Ангар
•	Горн

**Reply by Kor-Angar** [reply]
через zparser можно сделать телепорт
```D
    if (CurrentLevel != DRAGONISLAND_ZEN)
    {
        //CurrentLevel = DRAGONISLAND_ZEN;
        Wld_ChangeLevel("NEWWORLD\DRAGONISLAND.zen","SHIP_DECK_01");
    };
```
для корректного выхода из диалога надо использовать     `AI_StartState	(self, B_TeleportDragonisland,1,"");`  с "**1**"

пример кода для теста  у меня работает.
[B_TeleportDragonisland.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/9229833/B_TeleportDragonisland.zip)

**Comment by Kor-Angar** [comment]
```
func int StoryHelper_INFO1_Condition()
{
	return TRUE;
};
```

надо отключать машину времени. движения назад не работает полноценно и просто ломает игру. 

предлагаю в качестве условий в Condition поставить проверку на главу.

**Reply by dosinabox** [reply]
принято

**Comment by Kor-Angar** [comment]
я сначала тестил на яблоке. потом добавил сторихелперу. пример в архиве.зип   01.08.2022, 01:41, "dosinabox" ***@***.***>: Это для руны или ..?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by Kor-Angar** [comment]
при запуске 3 главы, можно менять мир на хоринос (NewWorld)
при запуске 4 главы, можно менять мир на миненталь (OldWorld)
